### DISCUSSION TITLE
[Plugin Request] Allow only the generation of scalars for types (for Prisma 2)

### DISCUSSION THREAD

**Discussion by Drakota** [dotansimha/graphql-code-generator#2813] [General]
**Is your feature request related to a problem? Please describe.**
Right now I'm trying to work with Prisma2 and Codegen together and I'm struggling with type clashing. The types generated by Codegen also includes relationships and custom types.
Ex. 
![image](https://user-images.githubusercontent.com/22665836/67318747-dfb87880-f4d9-11e9-991b-e396dd8e0f6d.png)

But when using Photon to fetch a course, the type looks like this:
![image](https://user-images.githubusercontent.com/22665836/67319654-396d7280-f4db-11e9-914c-fbb6e2764eb1.png)

**Describe the solution you'd like**
I know I can fix the string <-> Date difference by using the scalars option in the configuration, but I'd be nice to also have an option to generate only the scalars of a type to prevent this conflict.

This problem is especially prominent when dealing with resolvers.

Anyone had more luck using those two tools together?

**Comment by dotansimha** [comment]
@Drakota I think it will be best to have an alternative for the base `typescript` plugin, that generates TS types that match the structure that Prisma 2 expects. 

PRs are welcome ;)

**Comment by Drakota** [comment]
At the moment Prisma2 is still in beta and a moving target in each releases, that's why before going all out with a plugin for Codegen-Prisma2, I thought the config options `onlyGenerateScalars` would be a nice workaround. I'd be more than happy to contribute on this when Prisma2 gets more mature though.
